lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -347,9 +412,28 @@ void SessionStateEnforcer::m5g_update_session_state_to_amf(
                  << " with teid " << teid;
     return;
   }
-  bool amf_update_pending = false;
-  uint32_t cur_version;
   auto& session       = **session_it;
+  cur_version  = session->get_current_version();
+  if (version < cur_version ) {
+      MLOG(MINFO) << "UPF verions of session imsi " << imsi
+                  << " of  teid " << teid <<" recevied version " << version
+		  << " SMF latest version: " << cur_version << " Resending";
+      if (session->inc_throttle()) {

```

Comment:
is it possible to use `m5g_add_rules_and_update_upf` instead here. Or te crate a function that can be used in both `m5g_add_rules_and_update_upf` and `m5g_process_response_from_upf`. Both actions seems to be executing the same logic, so if so they should belong to one single function

Reply:
common logic  amalgmated, address the comment